query_and_submit.py

```diff
@@ -21,19 +21,49 @@
 metrics = {}
 
 for metric_name, query in config["queries"].items():
-  sql = query["countSql"]
-  condition_keyword = "and" if re.search("where", sql, re.IGNORECASE) else "where"
+  count_sql = query.get("countSql")
+  content_sql = query.get("contentSql")
   timestamp_col = query["timestampColumn"]
-  metrics[metric_name] = json.loads(subprocess.check_output(["psql", os.environ["DBURI"], "-Atc", f"{sql} {condition_keyword} \"{timestamp_col}\" between '{interval_start}' and '{interval_end}'"]))
-
-requests.post(
-  url=config["endpoint"],
-  json={
-    "startInterval": interval_start,
-    "endInterval": interval_end,
-    **metrics,
-  },
-  headers={
-    "xc-token": metrics_token,
-  }
-)
+  
+  if content_sql:
+    condition_keyword = "and" if re.search("where", content_sql, re.IGNORECASE) else "where"
+    db_result = subprocess.check_output(["psql", os.environ["DBURI"], "-Atc", f"{content_sql} {condition_keyword} \"{timestamp_col}\" between '{interval_start}' and '{interval_end}'"]).decode().split("\n")
+    db_result = filter(lambda x: x != "", db_result)
+
+    content_metric = {}
+    for result in db_result:
+      category, content, created_at = result.split("|")
+
+      content_metric["category"] = category
+      content_metric["content"] = content
+      content_metric["date"] = created_at
+
+      json_output = {
+        "createdAt": interval_start,
+        "endInterval": interval_end,
+        **content_metric
+      }
+
+      requests.post(
+        url=config["endpoint"],
+        json=json_output,
+        headers={
+          "xc-token": metrics_token,
+        }
+      )
+
+  else:
+    condition_keyword = "and" if re.search("where", count_sql, re.IGNORECASE) else "where"
+    metrics[metric_name] = json.loads(subprocess.check_output(["psql", os.environ["DBURI"], "-Atc", f"{count_sql} {condition_keyword} \"{timestamp_col}\" between '{interval_start}' and '{interval_end}'"]))
+
+    requests.post(
+      url=config["endpoint"],
+      json={
+        "startInterval": interval_start,
+        "endInterval": interval_end,
+        **metrics,
+      },
+      headers={
+        "xc-token": metrics_token,
+      }
+    )
```

Comment:
If this post is not inside the else, it will post empty record

Reply:
Ah, makes sense. However, if this is in the else it won't work to collect all other metrics in one entry as before.

There are two ways I could see how we can fix that: 
- Either remove that part completely and have two different jobs (one for content and one for count) using two different images.
- Or we use the bulk post method and only post once at the end.

We can also talk about it.